FirebaseAuth/Sources/User/FIRUser.m

```diff
@@ -1129,8 +1129,6 @@ - (void)linkWithEmailPassword:(FIREmailPasswordAuthCredential *)credential
                         callInMainThreadWithAuthDataResultAndError(completion, nil, keychainError);
                         return;
                       }
-                      [self signOutIfTokenIsInvalidWithError:error];
```

Comment:
Maybe this signOut should be in the if block above?

Reply:
I'm still working on this. Wait a bit. 😂

Reply:
Confirmed that it's not necessary. `updateKeychain` should not raise errors of invalid token.